docs/guides/ipsec-guide.md

```diff
@@ -0,0 +1,129 @@
+# IPsec Offload Guide
+
+## 1. Overview
+
+The [ipdk-io/ipsec-recipe](https://github.com/ipdk-io/ipsec-recipe) is an
+independent IPDK recipe, which provides strongSwan plugin code to configure and
+program control messages into the target.
+Consult the ipsec-recipe for full setup information.
+
+This document provides information pertaining to the gRPC server and the
+IPsec-specific feature details, which the P4 Control Plane provides. 
+
+The IPsec offload feature is only supported on the Intel&reg; IPU E2100 target.
+
+## 2. Feature Overview
+
+The `infrap4d` process provides the gRPC server-side support for P4RT and
+gNMI messages. strongSwan acts as the gRPC client in this context.
+Alternatively, you can use other clients that implement the IKE stack. The
+client communicates using P4RT to program Security Policy Database (SPD) and
+uses gNMI to configure Security Association Database (SAD) entries, which
+includes encryption keys, re-keying, soft and hard lifetime values etc.
+
+The YANG model uses for IPsec-offload is [available here](https://github.com/ipdk-io/openconfig-public/blob/master/release/models/ipsec/openconfig-ipsec-offload.yang).
+
+Follow the sequence of steps listed below for enabling IPsec functionality.
+
+### 2.1 Load Crypto Package
+
+Follow instructions in [Deploying P4 programs](https://github.com/ipdk-io/networking-recipe/blob/main/docs/guides/es2k/deploying-p4-programs.md)
+to load the hardware FXP pipeline with the crypto package.
+
+## 2.2 Configure and Run Infrap4d
+
+Follow instructions in [Running infrap4d on Intel&reg; IPU E2100](https://github.com/ipdk-io/networking-recipe/blob/main/docs/guides/es2k/running-infrap4d.md)
+guide to prepare system with following details.
```

Comment:
Can you say something more specific than "prepare [the] system with [the] following details?" Why should the user do this? What does the preparation accomplish?